File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    terms = polynomial.split(" + ")
    x_c = 0
    consts = 0

    for term in terms:
        if "x" in term:
            c = term.replace("x", "")
            if c == "":
                c = 1
            else:
                c = int(c)
            x_c += c
        else:
            consts += int(term)
    answer = ""
    if x_c:
        if x_c == 1:
            answer = "x"
        else:
            answer = f"{x_c}x"
        if consts:
            answer = answer + " + "
    if consts:
        answer = answer + str(consts)
    return answer

File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
  let x = 0;
  let n = 0;
  polynomial.split(' + ').forEach((v) => {
    if (v === 'x') {
      x += 1;
    } else if (v.includes('x')) {
      console.log(v);
      x += parseInt(v.replace('x', ''));
    } else {
      n += parseInt(v);
    }
  });
  const answer = [];
  if (x > 0) {
    if (x === 1) {
      answer.push('x');
    } else {
      answer.push(`${x}x`);
    }
  }
  if (n > 0) {
    answer.push(n);
  }
  return answer.join(' + ');
}